### PR TITLE
[generate] support return_log_probs for generate

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR Category

 Execute Infrastructure

### PR Types

Improvements

### Description

支持generate 接口传递return_log_probs 参数，如果是true 会返回log_probs 结果

### 是否引起精度变化

否

### CODE REVIEW

File: src/paddlefleet/generation/greedy_generator.py [hunk 1]
```diff
@@ -346,10 +347,19 @@ def generate(
...
-            Tensor of shape ``[B, L + num_generated]`` containing the
-            prompt plus generated tokens.
+            If ``return_log_probs=False``: Tensor of shape
+            ``[B, L + num_generated]`` containing the prompt plus generated
+            tokens.
+
+            If ``return_log_probs=True``: A tuple
+            ``(generated, output_log_probs)`` where ``generated`` is as above
+            and ``output_log_probs`` is a list of length ``B``, each element
+            being a list of per-step log-probs for the generated tokens only
```
Comment (risemeup1111) [inline src/paddlefleet/generation/greedy_generator.py:367]: P2 **优先级：P2**
这里新增了 `output_log_probs`，但语义还不够明确：当前实现是基于 repetition penalty 后的 raw logits 计算，还是希望和 `temperature/top_k/top_p` 的实际采样分布对齐。现在文档和测试都没有把这点说清，调用方很容易按“采样分布 log-prob”去理解而用错。
建议在这里补一句明确的定义，并补一个非默认采样参数的回归测试，把这条契约锁住。
**处理要求：请针对该评论进行回复（同意并已修改请回复 Done，不同意请说明理由）。**

**Review by risemeup1111 — COMMENTED** [review]
已复查，未发现需要阻塞合入的问题；关于 `return_log_probs` 的语义边界和测试覆盖，我留了一条 inline 建议，细节见评论。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: src/paddlefleet/generation/greedy_generator.py (hunk 1 above)
Comment (xiaoguoguo626807) [reply src/paddlefleet/generation/greedy_generator.py:367]: Done

Comment (risemeup1111) [reply src/paddlefleet/generation/greedy_generator.py:367]: 我这边复查当前 head（29c08546c8db2065b3ab632884b20a371d20a5d8），`return_log_probs` 相关的说明和测试还没有看到对应更新，所以这条暂时还不能按已处理算。
如果你已经补了，请再推一个新的 commit 后我继续复查；如果是决定保留当前实现，也请在这里说明一下取舍，我再判断是否可以接受。

File: tests/single_card_tests/model/test_greedy_generator.py [hunk 2]
```diff
@@ -490,6 +490,269 @@ def test_debug_mode_eos_stops_early(self):
...
+        gen.cache = DynamicKVCache(num_layers=1)
+
+        input_ids = paddle.to_tensor([[1]], dtype="int64")
+        _, log_probs = gen.generate(
+            input_ids,
+            max_new_tokens=1,
+            return_log_probs=True,
+            temperature=2.0,
+            top_k=5,
+        )
```
Comment (risemeup1111) [inline tests/single_card_tests/model/test_greedy_generator.py]: P1 **优先级：P1**
这个测试在 `temperature=2.0, top_k=5` 下会走 `paddle.multinomial` 随机采样。当前 logits 经过温度后是 `chosen=4.0`、其余 top-k token 为 `0.0`，因此仍有大约 6.8% 的概率采到非 `chosen_tok`；一旦采到其它 token，`log_probs[0][0]` 记录的就是其它 token 的 log-prob，下面与 `expected_pre_temp` 的断言会偶发失败，导致 CI flaky。
建议把采样结果 mock 成固定 token，仍然保留 `temperature/top_k` 这条路径，但去掉测试随机性，例如：
```suggestion
        from unittest.mock import patch

        input_ids = paddle.to_tensor([[1]], dtype="int64")
        with patch(
            "paddlefleet.generation.greedy_generator.paddle.multinomial",
            return_value=paddle.to_tensor([[chosen_tok]], dtype="int64"),
        ):
            _, log_probs = gen.generate(
                input_ids,
                max_new_tokens=1,
                return_log_probs=True,
                temperature=2.0,
                top_k=5,
            )
```
**处理要求：请针对该评论修复并提交新的 commit。**

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
已复查新提交：之前关于 `return_log_probs` 语义边界的建议已经通过文档和回归测试补上。当前还有一个新增测试的随机性问题需要修复，细节见 inline 评论；修复后我再继续复查。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: tests/single_card_tests/model/test_greedy_generator.py (hunk 2 above)
Comment (xiaoguoguo626807) [reply tests/single_card_tests/model/test_greedy_generator.py]: Done

**Review by risemeup1111 — APPROVED** [review]
已复查新提交，`return_log_probs` 的说明和回归测试都已经补齐，之前关注的随机性问题也已修掉，当前未见新的阻塞项。细节在之前的 inline 评论里。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: tests/single_card_tests/model/test_greedy_generator.py (hunk 2 above)
Comment (risemeup1111) [reply tests/single_card_tests/model/test_greedy_generator.py]: 已确认，当前 head 已通过固定 `paddle.multinomial` 去掉这个测试里的随机采样，我也已基于该状态给出通过意见。这里无需再补其它修改。

**Review by changeyoung98 — APPROVED** [review]
